### PR TITLE
Create SynapseSample.py

### PR DESCRIPTION
**Pull request by ManojRaheja**
Added Sample file for Synapse

#### Pull Request Description

Added Sample code for Synapse

---

#### Future Release Comment
Added sample pyspark code to demonstrate use of link service and integrated Kusto Spark connector in Azure Synapse. 

**Breaking Changes:**
- None

**Features:**
- Azure Data Explorer connector for Azure Synapse

**Fixes:**
- None

### CODE REVIEW

File: samples/src/main/python/SynapseSample.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Optional:
+sc._jvm.com.microsoft.kusto.spark.utils.KustoDataSourceUtils.setLoggingLevel("all")
+
+
+#######################
+# BATCH READ EXAMPLE  #
+#######################
+
+# Read data from Azure Data Explorer table(s)
+# In this example the Kusto Spark connector will determing the optimal path to get data: API for small data sets, Export/Distributed mode for large datasets.
```
Comment (ohadbitt) [inline samples/src/main/python/SynapseSample.py:10]:
```suggestion
# In this example the Kusto Spark connector will determine the optimal path to get data: API for small data sets, Export/Distributed mode for large datasets.
```

File: samples/src/main/python/SynapseSample.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Optional:
```
Comment (ohadbitt) [inline samples/src/main/python/SynapseSample.py:1]: Should we explain that this is not the repository that holds the synapse integration code? we could say that the other package utilizes this one and it is private

File: samples/src/main/python/SynapseSample.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            .load()
+
+display(kustoDf)    
+
+
+#######################
+# BATCH WRITE EXAMPLE #
+#######################
+
+# Write data to a Azure Data Explorer table
```
Comment (ohadbitt) [inline samples/src/main/python/SynapseSample.py:50]:
```suggestion
# Write data to an Azure Data Explorer table
```

File: samples/src/main/python/SynapseSample.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+kustoQ = dfStream \
+    .writeStream \
+    .format("com.microsoft.kusto.spark.synapse.datasink.KustoSynapseSinkProvider") \
+    .option("spark.synapse.linkedService", "<link service name>") \
+    .option("kustoDatabase", "<Database name>") \
+    .option("kustoTable", "<Table name>") \
+    .trigger(once = True)
+    
+kustoQ.start().awaitTermination(60*8)
```
Comment (ohadbitt) [inline samples/src/main/python/SynapseSample.py:122]: i think its better here to not have an await time - as usually when working from databricks/synapse we dont won't to stop the streaming job

**Review by ohadbitt — APPROVED** [review]

File: samples/src/main/python/SynapseSample.py (hunk 4 above)
Comment (ManojRaheja) [reply samples/src/main/python/SynapseSample.py:122]: As discussed, please update as you think fit.

**Pull request merged**